File: common/src/client/java/juuxel/adorn/client/ClientNetworkBridge.java
package juuxel.adorn.client;

import juuxel.adorn.util.InlineServices;
import juuxel.adorn.util.Services;
import net.minecraft.network.protocol.common.custom.CustomPacketPayload;

@InlineServices
public interface ClientNetworkBridge {
    void sendToServer(CustomPacketPayload payload);

    @InlineServices.Getter
    static ClientNetworkBridge get() {
        return Services.load(ClientNetworkBridge.class);
    }
}

File: common/src/main/kotlin/juuxel/adorn/client/ClientNetworkBridge.kt
package juuxel.adorn.client

import juuxel.adorn.menu.DataChannel
import juuxel.adorn.util.InlineServices
import juuxel.adorn.util.loadService
import net.minecraft.item.ItemStack

interface ClientNetworkBridge {
    fun sendSetTradeStack(syncId: Int, slotId: Int, stack: ItemStack)
    fun sendDataChannelToServer(channel: DataChannel<*>)

    @InlineServices
    companion object {
        private val instance: ClientNetworkBridge by lazy { loadService() }
        fun get(): ClientNetworkBridge = instance
    }
}
